hack/verify-govulncheck.sh

```diff
@@ -40,9 +40,9 @@ git worktree add -f "${WORKTREE}" "${BRANCH}"
 # Clean up the copy on exit
 kube::util::trap_add "git worktree remove -f ${WORKTREE}" EXIT
 
-govulncheck -scan module ./... > "${KUBE_TEMP}/head.txt"
+govulncheck -scan package ./... > "${KUBE_TEMP}/head.txt" || true
```

Comment:
why add || true?

Reply:
This is intentional. We want it to be non-breaking CI check

Reply:
For additional context, the behavior of govulncheck has changed from 1.0 to 1.1; it now returns nonzero if there are findings. Therefore @ArkaSaha30 hotwired that line of the script to always return true so that it won't block CI

Reply:
My reading is different from version 1.0.1 https://pkg.go.dev/golang.org/x/vuln@v1.0.1/cmd/govulncheck

> Govulncheck exits successfully (exit code 0) if there are no vulnerabilities, and exits unsuccessfully if there are. It also exits successfully if the -json flag is provided, regardless of the number of detected vulnerabilities.

what is the point of running the script if never fails ?

Reply:
Added more details here: https://github.com/kubernetes/kubernetes/pull/125863#issuecomment-2264194595

Reply:
1. I thought we wanted _symbol_-level scanning on release branches, not package... to know if we were actually using an impacted symbol?

2. Agree we don't want to fail CI on a vulnerability report, so `|| true` seems correct here currently. However, the description of this PR mentions support for `-format openvex`, which we're not actually using, and if we did, that would avoid the non-zero exit code ("It also exits successfully if the 'format -json' ('-json'), '-format sarif', or '-format openvex' is provided, regardless of the number of detected vulnerabilities.")

Reply:
Master branch PR is here: https://github.com/kubernetes/kubernetes/pull/125864